src/fetchers.ts

```diff
@@ -298,28 +297,60 @@ async function fetchHelm(
 }
 
 async function fetchNixpkgs(
-  spec: NixpkgsFetch,
+  _spec: NixpkgsFetch,
   context: FetchContext,
 ): Promise<Version[]> {
-  const version = await fetchUrl("https://channels.nixos.org/" + spec.channel, context, undefined, { redirect: "manual" }, async (response) => {
-    const regexp = /releases.nixos.org\/nixpkgs\/([^/]+)/;
-    if (response.status !== 301) {
-      throw new Error(
-        `Got status ${response.status}: ${await response.text()}`,
-      );
-    }
-    const location = response.headers.get("Location");
-    if (location === null) {
-      throw new Error(`Could not get location for ${response.url}`);
-    }
-    const match = regexp.exec(location);
-    if (match === null) {
-      throw new Error(`Could not parse ${response.url}`);
+  const cacheKey = "nix-channel:nixpkgs-unstable";
+  let versions: string[] | undefined = undefined;
+  if (!context.update) {
+    const cached = context.cache.getBody(cacheKey);
+    if (cached) {
+      versions = JSON.parse(cached);
     }
-    return match[1];
-  });
+  }
+
+  if (versions === undefined) {
+    await dom.initParser();
+    versions = [];
+
+    let continuationToken: string | null = null;
+    do {
+      const url = new URL("https://nix-releases.s3.amazonaws.com");
+      url.searchParams.set("list-type", "2");
+      url.searchParams.set("prefix", "nixpkgs/nixpkgs-");
+      url.searchParams.set("delimiter", "/");
+      if (continuationToken !== null) {
+        url.searchParams.set("continuation-token", continuationToken);
+      }
+
+      const response = await fetch(url);
+      if (response.status !== 200) {
+        throw new Error(
+          `Got status ${response.status}: ${await response.text()}`,
+        );
+      }
+
+      const rawText = await response.text();
+      const doc = new dom.DOMParser().parseFromString(rawText, "text/html");
```

Comment:
Apprently it's not html document, but since deno_dom library works only with this MIME (I tried `application/xml`, and it's not working), and we dont want to import another XML parsing library, this might be fine.

Reply:
Indeed, we already use it to parse an RSS feed (jenkins updates) and it just worked so I just reused it here.